File: C++/Calc.cpp
#include <iostream>
#include <cmath> // pow

#include "Calc.h"

Calc::Calc()
    : x_new_()
{}

double Calc::operator()(std::function<double(double)>f, std::function<double(double)>df, double x0, int max_iter, double epc)
{
    double x = x0;
    int iter = 0;
    while(1)
    {
        x_new_ = x - f(x)/df(x);
        if (std::abs(x - x_new_) < epc)
        {
            break;
        }
        x = x_new_;
        iter ++;
        if (iter == max_iter)
        {
            break;
        }
    }
    return x_new_;
}

File: Python/calc.py
class Calc:
    def __init__(self):
        self.eps = 1e-10
        self.max_iter = 1000

    def newton_func(self, f, df):
        def newton(x0):
            x = x0
            iter = 0
            while True:
                x_new = x - f(x)/df(x)
                if abs(x-x_new) < self.eps:
                    break
                x = x_new
                iter += 1
                if iter == self.max_iter:
                    break
            return x_new
        return newton
